ddsrecorder_participants/include/ddsrecorder_participants/recorder/size_tracker/McapSizeTracker.hpp

```diff
@@ -0,0 +1,197 @@
+
+    /**
+     * @brief Get space needed to write channel
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_channel_size(
+            const mcap::Channel& channel,
+            const uint64_t& size_metadata);
+
+    /**
+     * @brief Get space needed to write blank channel
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_blank_channel_size(
+            const std::string& channel_name);
+
+    /**
+     * @brief Get space needed to write attachment
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_attachment_size(
+            const std::unique_ptr<fastrtps::rtps::SerializedPayload_t>& serialized_payload);
+
+    /**
+     * @brief Check if there is enough space to write the current file
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void decrease_mcap_size(
+            const std::uint64_t& size);
+
+    /**
+     * @brief Check if there is enough space to write the current file
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_and_increase_mcap_size_final(
+            const std::uint64_t& size);
+
+    /**
+     * @brief Check if there is enough space to write the current file when adding a message
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_and_increase_mcap_size(
+            const uint64_t& msg_datasize);
+
+    /**
+     * @brief Check if there is enough space to write the current file when adding a schema
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_and_increase_mcap_size(
+            const mcap::Schema& schema);
+
+    /**
+     * @brief Check if there is enough space to write the current file when adding a channel
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_and_increase_mcap_size(
+            const mcap::Channel& channel,
+            const uint64_t& size_metadata);
+
+    /**
+     * @brief Check available space on disk
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_available_space(
+        const std::string& file_path);
+
+protected:
+
+    //! Total file size
+    std::uint64_t mcap_size_{MCAP_FILE_OVERHEAD}; // MCAP file size is initialized with MCAP_FILE_OVERHEAD
+
+    //! Space available in disk
+    std::uintmax_t space_available_when_open_;
+
+    //! Boolean that describes if disk is full or not
+    bool disk_full_ = false;
+
+    //! MCAP file overhead
+    /**
+     * To reach this number, we use the following constants:
+     *   - Header + Write Header = 18
+     *   - Metadata + Write Metadata + Write MetadataIndex = 75 + 24 + 36
+     *   - Write ChunkIndex = 73
+     *   - Write Statistics = 55
+     *   - Write DataEnd + Write SummaryOffSets = 13 + 26*6
+     */
+    static constexpr std::uint64_t MCAP_FILE_OVERHEAD{450};
+
+    //! Additional overhead size for a MCAP message
+    static constexpr std::uint64_t MCAP_MESSAGE_OVERHEAD{31 + 8 + 8}; // Write Message + TimeStamp + TimeOffSet
```

Comment:
NIT: maybe it would be better to split the sum in different constants. Perhaps some of these could be moved to the methods making use of them.

Reply:
Suggestion disregarded.